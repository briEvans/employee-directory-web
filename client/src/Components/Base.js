import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Search from './Search';

import '../App.scss';

class Base extends Component {
    render() {
      return (
        window.location.pathname === '/' ?
          (<div className="app-body">
          <span className="welcome">
            <p>Hello Admin User. This is a reminder that as an ADMIN user, you have the ability to <span class="colorTurq">create and remove employees</span></p></span>
          <h1>Employee Directory</h1>
          <Search />
          <ul className="nav">
            <li>
              <Link className="app-link bgTurq" to="/employees" >
                <span className="action">List Employees</span>
                <p> See all employees listed in alphabetical order by name</p>
              </Link>
            </li>
            <li>
              <Link className="app-link bgGrass" to="/departments">
                <span className="action">List Departments</span>
              <p> See all Departments listed in alphabetical order in the company</p>
            </Link>
            </li>
            <li>
              <Link className="app-link bgLime" to="/locations">
                <span className="action">List Locations</span>
                <p>See all Locations listed in alphabetical order in the company</p>
                </Link>
            </li>
          </ul>
        </div>)
      :
        (<div className="bg"></div>)
    );
  };
};

export default Base;
