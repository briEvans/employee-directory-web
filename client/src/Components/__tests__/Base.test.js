import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route} from 'react-router-dom';
import { mount, shallow, render } from 'enzyme';

import Base from '../Base';

describe("Base component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    <Router>
      ReactDOM.render(<Base/>, div);
    </Router>
    ReactDOM.unmountComponentAtNode(div);
  });

  test("it matches the snapshot", () => {
    const component = shallow(<Base />);
    expect(component).toMatchSnapshot();
  });
});
