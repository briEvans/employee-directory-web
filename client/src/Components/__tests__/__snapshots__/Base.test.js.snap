// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base component it matches the snapshot 1`] = `
<div
  className="app-body"
>
  <h1>
    Employee Directory
  </h1>
  <div
    className="search search-large"
  >
    <form
      className="product-form"
      onSubmit={[Function]}
    >
      <input
        placeholder="John Doe..."
        type="text"
      />
      <button>
        Go
      </button>
    </form>
  </div>
  <ul
    className="app-nav"
  >
    <li>
      <Link
        className="app-link bgTurq"
        to="/employees"
      >
        <span
          className="action"
        >
          List Employees
        </span>
        <p>
           See all employees listed in alphabetical order by name
        </p>
      </Link>
    </li>
    <li>
      <Link
        className="app-link bgGrass"
        to="/locations"
      >
        <span
          className="action"
        >
          List Locations
        </span>
        <p>
          See all Locations listed in alphabetical order in the company
        </p>
      </Link>
    </li>
    <li>
      <Link
        className="app-link bgLime"
        to="/departments"
      >
        <span
          className="action"
        >
          List Departments
        </span>
        <p>
           See all Departments listed in alphabetical order in the company
        </p>
      </Link>
    </li>
  </ul>
</div>
`;
